errors.go

```diff
@@ -0,0 +1,23 @@
+package customerio
+
+import (
+	"fmt"
+	"log"
+)
+
+// CustomerIOError is returned by any method that fails at the API level
+type CustomerIOError struct {
+	status int
+	url    string
+	body   []byte
+}
+
+func (e *CustomerIOError) Error() string {
+	return fmt.Sprintf("%v: %v %v", e.status, e.url, string(e.body))
+}
+
+func CheckErr(err error) {
+	if err != nil {
+		log.Fatal("action failed: ", err)
```

Comment:
Calling log.Fatal() in a common error checking function breaks our API contract with callers.

Each API resource function in this library follows the same signature:
```
func (c *CustomerIO) Resource(args...) error {}
```

This tells consumers of the API library that each API request they call through the library will return an `error` for them to handle. By instead calling `log.Fatal()`we'll cause unexpected failures in the programs depending on this library since that function calls `os.Exit(1)` under the hood terminating the calling program.

Reply:
@joepurdy I strongly agree with you. This was totally my bad. I mistakenly do `log.Fatal()` my plan was to use `log.Println()`. And I like to keep the `errors` package separate from my any other package. I am looking forward to hearing from you.